epoll/src/main/scala/fs2/io/epoll/unsafe/EpollSystem.scala

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright 2023 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.io.epoll
+package unsafe
+
+import cats.effect.{IO, Resource}
+import cats.effect.std.Mutex
+import cats.syntax.all._
+import cats.effect.unsafe.PollingSystem
+
+import java.io.IOException
+import java.util.concurrent.ConcurrentHashMap
+
+import jnr.ffi.Memory;
+import jnr.constants.platform.Errno;
+
+import scala.annotation.tailrec
+
+object EpollSystem extends PollingSystem {
+
+  import epoll._
+  import eventfd._
+  import libc.jnr._
+
+  private[this] final val MaxEvents = 64
+
+  type Api = FileDescriptorPoller
+
+  def close(): Unit = ()
+
+  def makeApi(register: (Poller => Unit) => Unit): Api =
+    new FileDescriptorPollerImpl(register)
+
+  def makePoller(): Poller = {
+    val epfd = epoll_create1(0)
+    if (epfd == -1)
+      throw new IOException(strerror(errno()))
+
+    val evfd = libc.jnr.eventfd(0, EFD_SEMAPHORE | EFD_NONBLOCK)
```

Comment:
Oh. I misunderstood the spec, you are correct.

If we set `EFD_SEMAPHORE`, the counter is decremented by 1 instead of being set to 0, so it is not suitable here.